### DISCUSSION TITLE
Display custom attribute to user while using select2_from_ajax field with Fetch and InlineCreation

### DISCUSSION THREAD

**Discussion by GregaUNK** [Laravel-Backpack/community-forum#1147] [Q&A (Help)] [answered]
Hi,

In the database, I have two fields: contact_name and contact_phone. I want users to see “contact_name (contact_phone)” when selecting in select2_from_ajax. I’ve also enabled InlineCreation for the same field.

By default, I couldn’t find a solution to use my full_contact_name attribute, which is defined in the model.

I worked around this by using custom queries.

For fetching:

```

    protected function fetchPartnerContact()
    {
        $partnerId = backpack_form_input();
        if (!isset($partnerId['partner_id'])) {
            return false;
        }
        $partnerId = $partnerId['partner_id'];

        return $this->fetch([
            'model' => \App\Models\PartnerContact::class,
            'searchable_attributes' => ['contact_name', 'contact_phone'],
            'paginate' => 20,
            'searchOperator' => 'LIKE',
            'query' => function ($model) use ($partnerId) {
                return $model->select([
                    'id',
                    'contact_name',
                    'contact_phone',
                    \DB::raw("CONCAT(contact_name, ' (', contact_phone, ')', contact_phone) as full_contact_name")
                ])->where('partner_id', $partnerId);
            }
        ]);
    }
```

For inline creation:
```
 public function storeInlineCreate()
    {
        // Call the inline create trait to save the contact
        $response = $this->inlineTraitStore();

        // Decode the JSON response to access the data
        $responseData = $response->getData();

        // Fetch the newly created PartnerContact using the ID from the response
        $newContact = \App\Models\PartnerContact::find($responseData->data->id);

        // Add the concatenated full_contact field to the response data
        $responseData->data->full_contact_name = $newContact->contact_name . ' (' . $newContact->contact_phone . ')';

        // Return the modified JSON response
        return response()->json($responseData);
    }
```

This works, but I’d like to ask if this is an appropriate solution? Or is there a better, smoother way to achieve the same result?

**Comment by pxpm** [comment, marked as answer]
Hey @GregaUNK I think you will be better using Laravel Accessors/Mutators https://laravel.com/docs/11.x/eloquent-mutators#accessors-and-mutators and then add that as the select `attribute` in the field.

Basically define a `public function fullContactName(): Attribute` in your model and in the field, define `attribute => fullContactName`. Don't forget you need to add that attribute to model `$appends`. https://laravel.com/docs/11.x/eloquent-serialization#appending-values-to-json

I think adding the attribute to `$appends` it's what missing in your use case, and non of those overwrites would be needed. 

Cheers

**Reply by GregaUNK** [reply]
Thanks! 

Totally correct - $appends was missing in my puzzle :)
